File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        res = 0
        streak = 1
        
        nums.sort()
        for i in range(len(nums) - 1):
            if nums[i] + 1 == nums[i + 1]:
                streak += 1
            elif nums[i] != nums[i + 1]:
                res = max(res, streak)
                streak = 1

        res = max(res, streak)
        return res

        

File: Data Structures & Algorithms/three-integer-sum/submission-3.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> res = new ArrayList<>();

        for (int i = 0; i < nums.length; i++){
            int a = nums[i];
            if (i > 0 && nums[i-1] == a){
                continue;
            }

            int l = i + 1;
            int r = nums.length - 1;

            while(l < r){
                int threeSum = a + nums[l] + nums[r];
                if (threeSum > 0){
                    r--;
                }else if (threeSum < 0){
                    l++;
                }else{
                    res.add(Arrays.asList(a, nums[l], nums[r]));
                    l++;
                    r--;
                    while (l < r && nums[l] == nums[l-1]){
                        l++;
                    }
                }
            }
        }

        return res;
    }
}
